tests/phpunit/tests/media.php

```diff
@@ -3710,6 +3710,86 @@ function() {
 		$this->assertSame( 3, $omit_threshold );
 	}
 
+	/**
+	 * Tests that wp_get_loading_attr_default() returns the expected loading attribute value before loop but after get_header.
+	 *
+	 * @ticket 58211
+	 *
+	 * @covers ::wp_get_loading_attr_default
+	 *
+	 * @dataProvider data_wp_get_loading_attr_default_before_and_no_loop
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_attr_default_before_loop( $context ) {
+		global $wp_query, $wp_the_query;
+
+		$wp_query = new WP_Query( array( 'post__in' => array( self::$post_ids['publish'] ) ) );
+		$this->reset_content_media_count();
+		$this->reset_omit_loading_attr_filter();
+
+		// Lazy if not main query.
+		$this->assertSame( 'lazy', wp_get_loading_attr_default( $context ) );
+
+		// Set current query as main query.
+		$wp_the_query = $wp_query;
+
+		do_action( 'get_header' );
+		$this->assertFalse( wp_get_loading_attr_default( $context ) );
+
+		while ( have_posts() ) {
+			the_post();
+		}
+		$this->assertSame( 'lazy', wp_get_loading_attr_default( 'wp_get_attachment_image' ) );
+	}
+
+	/**
+	 * Tests that wp_get_loading_attr_default() returns the expected loading attribute if no loop.
+	 *
+	 * @ticket 58211
+	 *
+	 * @covers ::wp_get_loading_attr_default
+	 *
+	 * @dataProvider data_wp_get_loading_attr_default_before_and_no_loop
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_attr_default_no_loop( $context ) {
+		global $wp_query, $wp_the_query;
+
+		$wp_query = new WP_Query( array( 'post__in' => array( self::$post_ids['publish'] ) ) );
+		$this->reset_content_media_count();
+		$this->reset_omit_loading_attr_filter();
+
+		// Set current query as main query.
+		$wp_the_query = $wp_query;
+
+		// Ensure header is called already.
+		do_action( 'get_header' );
+
+		// if footer called before loop.
+		add_action(
+			'get_footer',
+			function() use ( $context ) {
+				$this->assertSame( 'lazy', wp_get_loading_attr_default( $context ) );
+			}
+		);
```

Comment:
Related to my above comment, this should not be needed.

Reply:
I have removed it.